src/Abp/BackgroundJobs/EventTriggerAsyncBackgroundJob.cs

```diff
@@ -0,0 +1,27 @@
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Events.Bus;
+
+namespace Abp.BackgroundJobs
+{
+    public class EventTriggerAsyncBackgroundJob<T> : AsyncBackgroundJob<T>, ITransientDependency
+        where T:EventData
+    {
+        public EventTriggerAsyncBackgroundJob(IEventBus eventBus)
+        {
+            
+            EventBus =eventBus ?? NullEventBus.Instance;
+        }
+
+        
+        /// <summary>
+        /// this property is not injected, so we have to inject it in constructor
+        /// </summary>
+        public IEventBus EventBus { get; set; }
```

Comment:
You are using property injection, no need to use constructor injection. Use

EventBus = NullEventBus.Instance;

in the constructor.

Reply:
when I register the generic type, property injection works